src/wblib/services/get_figures.py

```diff
@@ -14,46 +13,51 @@
 Image = Union[img.Image, Figure]
 
 
-def generate_external_figures(current_location: str,
-                              current_time: pd.Timestamp,
-                              logger: Callable)-> dict[str, Image]:
-    figures = dict()
+def generate_external_figures(
+    current_location: str,
+    current_time: pd.Timestamp,
+    logger: Callable
+) -> Iterator[tuple[str, Image]]:
     for product, function in EXTERNAL_PLOTS.items():
         if function is None:
             _warn_function_is_not_defined(product, logger)
             continue
         try:
             if product == 'ifs_meteogram':
-                figures[product] = function(current_location, current_time)
+                figure = function(current_location, current_time)
             else:
-                figures[product] = function(current_time)
+                figure = function(current_time)
+            yield (product, figure)
         except Exception as error:
-            msg = (f"Can not generate {product} with '{current_time}'. "
-                   "Please provide it manually or debug the code.")
+            msg = (
+                f"Can not generate {product} with '{current_time}'. "
+                "Please provide it manually or debug the code."
+            )
             logger(msg, "ERROR")
             print(error)
-    return figures
+            continue
 
 
-def generate_internal_figures(current_time: pd.Timestamp,
-                              logger: Callable) -> dict[str, dict[str, Image]]:
-    figures = dict()
+def generate_internal_figures(
+    current_time: pd.Timestamp, logger: Callable
+) -> Iterator[tuple[str, str, Image]]:
```

Comment:
@lkluft, making this functions generators is perhaps the biggest change. Asking for you review here as I have not use generators many times before. Thanks!

Reply:
I think the use of a generator makes sense here, as it simplifies the code around the function 👍 As far as I can tell, the implementation look good too 👌

Reply:
Thanks Lukas! Merging now to main.